inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_squeeze.cpp

```diff
@@ -24,7 +24,7 @@ void dynamicToStaticShapeSqueeze(std::shared_ptr<ngraph::Node> target) {
                      target, ngraph::opset3::Squeeze::type_info);
 
     const auto dsr = target->input_value(0).get_node_shared_ptr();
-    VPU_THROW_UNLESS(std::dynamic_pointer_cast<ngraph::vpu::op::DynamicShapeResolver>(dsr),
+    VPU_THROW_UNLESS(ngraph::is_type<ngraph::vpu::op::DynamicShapeResolver>(dsr),
```

Comment:
Why `dynamic_pointer_cast` does not work? 

CC @ilyachur

Reply:
Looks like it is an internal VPU node has some issues with RTTI?
Am I right? Or in other case should `dynamic_pointer_cast` and `is_type` should produce the same result.

Reply:
It looks like this problem only occurs on MacOS, and cases segmentation fault, but `is_type` works fine
I've tried to directly turn on RTTI, but it didn't help

Reply:
RTTI is enabled by default. We had the same problems when symbols aren't exported from the library. I am not sure that it is your case, need to check.

Reply:
How did you discovered your problem?

Reply:
@PoliOwl the issue usually happens when you use dynamic cast to a type which is located in a different shared library (in this case class must be export from that library). E.g. I believe if the issue comes when you cast to ngraph type which is not exported, but these types are located inside your library and the issue should not happen...

Try to build with clang and `-fsanitize=vptr` (see https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html) and run this test.